homeassistant/components/remember_the_milk.py

```diff
@@ -0,0 +1,249 @@
+        if token:
+            _LOGGER.debug("found token for account %s", account_name)
+            _create_instance(
+                hass, account_name, api_key, shared_secret, token,
+                stored_rtm_config, component, descriptions)
+        else:
+            _register_new_account(
+                hass, account_name, api_key, shared_secret,
+                stored_rtm_config, component, descriptions)
+
+    _LOGGER.debug("Finished adding all Remember the milk accounts")
+    return True
+
+
+def _create_instance(hass, account_name, api_key, shared_secret,
+                     token, stored_rtm_config, component, descriptions):
+    entity = RememberTheMilk(account_name, api_key, shared_secret,
+                             token, stored_rtm_config)
+    component.add_entity(entity)
+    hass.services.async_register(
+        DOMAIN, '{}_create_task'.format(account_name), entity.create_task,
+        description=descriptions.get(SERVICE_CREATE_TASK),
+        schema=SERVICE_SCHEMA_CREATE_TASK)
+
+
+def _register_new_account(hass, account_name, api_key, shared_secret,
+                          stored_rtm_config, component, descriptions):
+    from rtmapi import Rtm
+
+    request_id = None
+    configurator = hass.components.configurator
+    api = Rtm(api_key, shared_secret, "write", None)
+    url, frob = api.authenticate_desktop()
+    _LOGGER.debug('sent authentication request to server')
+
+    def register_account_callback(_):
+        """Callback for configurator."""
+        api.retrieve_token(frob)
+        token = api.token
+        if api.token is None:
+            _LOGGER.error('Failed to register, please try again.')
+            configurator.notify_errors(
+                request_id,
+                'Failed to register, please try again.')
+            return
+        else:
+            stored_rtm_config.set_token(account_name, token)
+            _LOGGER.debug('retrieved new token from server')
+
+            _create_instance(
+                hass, account_name, api_key, shared_secret, token,
+                stored_rtm_config, component, descriptions)
+
+            configurator.request_done(request_id)
+
+    request_id = configurator.async_request_config(
+        '{} - {}'.format(DOMAIN, account_name),
+        callback=register_account_callback,
+        description='You need to log in to Remember The Milk to' +
+        'connect your account. \n\n' +
+        'Step 1: Click on the link "Remember The Milk login"\n\n' +
+        'Step 2: Click on "login completed"',
+        link_name='Remember The Milk login',
+        link_url=url,
+        submit_caption="login completed",
+    )
+
+
+class RememberTheMilkConfiguration(object):
+    """Internal configuration data for RememberTheMilk class.
+
+    This class stores the authentication token it get from the backend.
+    """
+
+    def __init__(self):
+        """Create new instance of configuration."""
+        if os.path.isfile(CONFIG_FILE_NAME):
```

Comment:
This is wrong. You need to run every file through `hass.config.path(CONFIG_FILE_NAME)` so you get the path inside the config dir.

Reply:
Also use guard clauses:

```python
if not os.path.isfile(…):
    self._config = {}
    return

try:
```

Reply:
OK

Reply:
OK